### PR TITLE
layout: resize the filesystem after expanding it

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

**Comment by Itxaka** [comment]
@davidcassany could you have a look at this? I think its the correct approach to the resizing but Im not sure about it.

**Review by davidcassany — CHANGES_REQUESTED** [review]
This is correct but not complete.

Note this plugin supports ext[2-4] and xfs filesystems. So in case of an XFS partition it should call `xfs_growfs -d <device>`. You can use `blkid` tool to detect the underlaying filesystem.

I included XFS support mostly because I see this plugin creating the persistent partition which is likely to make use of a modern filesystem such as XFS.

Finally in addition, to make this changes effective in cOS note that immutable rootfs must include any new executable that layout plugin might call (`resize2fs`, `xfs_growfs`). See:
https://github.com/rancher-sandbox/cOS-toolkit/blob/0b3d79b22e5b09644fa91bf2253a5bf3252a5eec/packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh#L29-L32

**Comment by Itxaka** [comment]
@davidcassany @mudler ready for review!

File: pkg/plugins/layout.go [hunk 1]
```diff
@@ -387,13 +388,69 @@ func (dev *Disk) ExpandLastPartition(size uint, console Console) (string, error)
...
+		if err != nil {
+			return out, err
+		}
+		out, err = console.Run(fmt.Sprintf("resize2fs %s", device))
+
+		if err != nil {
+			return out, err
+		}
+	case "xfs":
+		// to grow an xfs fs it needs to be mounted :/
```
Comment (davidcassany) [inline pkg/plugins/layout.go:423]: Ouch, this is unfortunate

**Review by davidcassany — APPROVED** [review]
LGTM

**Review by mudler — APPROVED** [review]

**Pull request merged**